commit 57027e4a33f5e15d9dd3a1b3363d65b84a8b253c
Author: Yang Bo <yangbo@linuxdeepin.com>
Date:   Sun Jan 3 20:09:06 2016 +0800

    UBUNTU: SAUCE: bcache: prevent crash on changing writeback_running
    
    BugLink: http://bugs.launchpad.net/bugs/1357295
    
    commit a664d0f05a2ec02c8f042db536d84d15d6e19e81
        bcache: fix crash on shutdown in passthrough mode
    
    added a safeguard in the shutdown case. At least while not being
    attached it is also possible to trigger a kernel bug by writing into
    writeback_running. This change  adds the same check before trying to
    wake up the thread for that case.
    
    BugLink: http://bugs.launchpad.net/bugs/1357295
    
    (sent upstream and accepted into developer tree but not, yet Linus)
    Signed-off-by: Stefan Bader <stefan.bader@canonical.com>
    Acked-by: Seth Forshee <seth.forshee@canonical.com>
    Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
    Signed-off-by: Yang Bo <yangbo@linuxdeepin.com>

diff --git a/drivers/md/bcache/writeback.h b/drivers/md/bcache/writeback.h
index 0a9dab1..073a042 100644
--- a/drivers/md/bcache/writeback.h
+++ b/drivers/md/bcache/writeback.h
@@ -63,7 +63,8 @@ static inline bool should_writeback(struct cached_dev *dc, struct bio *bio,
 
 static inline void bch_writeback_queue(struct cached_dev *dc)
 {
-	wake_up_process(dc->writeback_thread);
+	if (!IS_ERR_OR_NULL(dc->writeback_thread))
+		wake_up_process(dc->writeback_thread);
 }
 
 static inline void bch_writeback_add(struct cached_dev *dc)
